### PR TITLE
Remove two GetImages functions from API

### PR DESCRIPTION
**Pull request by rhatdan**
[NO NEW TESTS NEEDED] This is just code cleanup.

The remote API has three different GetImages functions, which I believe
can be handled by just one function.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@jwhonce @flouthoc Looks like their is something special in the Docker API for handling dangling images.

I went through the code to make sure libpod, docker-compat and local all use the same code paths.

podman --remote images -a
podman images -a
DOCKER_HOST=unix:///run/user/3267/podman/podman.sock docker images -a

Are all passing exactly the same params and back the same content, but podman lists the <none> <none> images but the docker client does not. Any ideas?

https://github.com/containers/podman/issues/12567

File: pkg/api/handlers/compat/images.go
Comment (jwhonce) [inline pkg/api/handlers/compat/images.go:420]: By skipping the call to `ImageToImageSummary`:
* `ImageSummary.VirtualSize` will be incorrect.
* It appears `ID` will no longer be prefixed with `sha256:`
* `ImageSummary.RepoDigests` will no longer be formatted  rather the raw data will be returned.
* ...

**Review by jwhonce — CHANGES_REQUESTED** [review]
Compatibility needs to be doubled checked, missing call that populated/formatted fields.

File: pkg/api/handlers/compat/images.go
Comment (flouthoc) [inline pkg/api/handlers/compat/images.go:416]: For `compat` api we were ignoring `manifest` list I think merging this with common code would start printing `manifest` lists as well. 

I have only looked at the code yet. But I'll need to try this out inoder to make it sure that we are not breaking any existing behavior.

Comment (rhatdan) [reply pkg/api/handlers/compat/images.go:416]: This filter
+               filterList = append(filterList, "manifest=false")

Should remove all manifest images for the DockerAPI.

File: pkg/api/handlers/compat/images.go
Comment (rhatdan) [reply pkg/api/handlers/compat/images.go:420]: I have fixed the sha256.
VirtualSize seems to be the same between the two calls. (It is always just size).
I switched libpod output for RepoDigests to match Dockers.
I did not see any others.

**Comment by rhatdan** [comment]
@vrothberg PTAL at the error.
containers/common has changed the behavior of 

```
mkdir -p /tmp/test
podman pull alpine
podman push alpine dir:/tmp/test/
podman rmi  alpine
podman pull dir:/tmp/test
```

In current podman we get an image named
```
podman images
localhost/tmp/test                 latest      c059bfaa849c  4 weeks ago   5.87 MB
```

With the latest containers/common we get
```
podman images
<none>                <none>      c059bfaa849c  4 weeks ago   5.87 MB
```

I think the second option looks correct, and we should fix the pull_test.go code, but want to verify with  you.

**Comment by vrothberg** [comment]
@rhatdan, I agree that the second one is correct. I should have check the tests in Podman and Buildah after it has merged. I will re-apply the PR in c/common and then open PRs in Podman/Buildah.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@vrothberg @flouthoc @giuseppe PTAL

**Review by flouthoc — APPROVED** [review]

File: pkg/domain/infra/abi/images.go
Comment (Luap99) [inline pkg/domain/infra/abi/images.go:74]: I think this should be removed, @vrothberg correct?

Comment (flouthoc) [reply pkg/domain/infra/abi/images.go:74]: @Luap99 I think these are dangling images from `multistage` builds. I dont think API should return these. But I can check what docker is returning.

Comment (Luap99) [reply pkg/domain/infra/abi/images.go:74]: I mean the change should be removed, see https://github.com/containers/common/pull/879/commits/134e83ff55ce83a37756ef473bf083b65e2c2c3c
@vrothberg fixed the problem in c/common so this is no longer needed

Comment (flouthoc) [reply pkg/domain/infra/abi/images.go:74]: Ah i see. Fair if that is fixed in upstream then check could be removed.

Comment (vrothberg) [reply pkg/domain/infra/abi/images.go:74]: Yes, that's not needed anymore.

**Review by vrothberg — COMMENTED** [review]
Besides @Luap99's nit LGTM. Less (redundant) code is always a great thing!

File: pkg/domain/infra/abi/images.go
Comment (Luap99) [inline pkg/domain/infra/abi/images.go:63]: Please remove the change here and also remove the `filterPrefix` function above

Comment (rhatdan) [reply pkg/domain/infra/abi/images.go:63]: We are blocking double duplicate filters.  So if it has containers=external from some were else, then we don't want to add it again.

Comment (Luap99) [reply pkg/domain/infra/abi/images.go:63]: @vrothberg fixed this in c/common see the comments above, you have to rebase this to get the latest common

**Comment by TomSweeneyRedHat** [comment]
Could you touch up the title plase @rhatdan?  I think you meant "two", not "to".

File: pkg/domain/infra/abi/images.go
Comment (flouthoc) [inline pkg/domain/infra/abi/images.go:67]: This the same check where @Luap99 raised concern before. If this is handled by `c/common` then we might not needed it. But I don't think there is any blocker if we are checking for duplicates.

so LGTM

Comment (flouthoc) [reply pkg/domain/infra/abi/images.go:67]: @vrothberg @Luap99 ^

**Review by flouthoc — APPROVED** [review]
LGTM

File: pkg/domain/infra/abi/images.go
Comment (vrothberg) [reply pkg/domain/infra/abi/images.go:67]: Yes, the dangling-logic can/should be removed.

Comment (rhatdan) [reply pkg/domain/infra/abi/images.go:67]: The dangling check is looking for dupicates.  This function can be called in multiple different ways.  And certain code paths and Remote API can pass in a filter with dangling=true already set, which ends up adding the dangling=true, which I am trying to prevent.  If you are suggesting the common defaults to dangling=true?

Comment (vrothberg) [reply pkg/domain/infra/abi/images.go:67]: The code in question was added to work around the duplicate checks in c/common. In the meantime, c/common accepts duplicates as long as they use the same value, so we don't need the custom check here anymore.

**Comment by rhatdan** [comment]
@vrothberg @Luap99 Mergeme.

**Pull request merged**
